src/batchUploader.ts

```diff
@@ -334,13 +327,8 @@ export class BatchUploader {
             this.batchesQueuedForProcessing.unshift(...batchesThatDidNotUpload);
         }
 
-        // Update Offline Storage with current state of batch queue
-        if (!useBeacon && this.offlineStorageEnabled && this.batchVault) {
-            // Note: since beacon is "Fire and forget" it will empty `batchesThatDidNotUplod`
-            // regardless of whether the batches were successfully uploaded or not. We should
-            // therefore NOT overwrite Offline Storage when beacon returns, so that we can retry
-            // uploading saved batches at a later time. Batches should only be removed from
-            // Local Storage once we can confirm they are successfully uploaded.
+        if (this.offlineStorageEnabled && this.batchVault) {
```

Comment:
just checking - we shouldn't do this if !useBeacon right? why remove that clause

Reply:
I think Sam is correct here.  We _want_ to save when useBeacon is false.

Reply:
I removed it so that regardless if beacon is fired, we purge local storage to prevent duplicates.  I initially put it in so that we avoid touching local storage if beacon is fired. I'll put it back.